perl6.html.markdown

```diff
@@ -798,19 +798,39 @@ class Item does PrintableVal {
 
 ### Exceptions
 # Exceptions are built on top of classes, in the package `X` (like `X::IO`).
-# You can access the last exception with the special variable `$!`
-# (use `$_` in a `CATCH` block) Note: This has no relation to $!variables.
+# In Perl6 exceptions are automatically 'thrown'
+open 'foo' #> Failed to open file foo: no such file or directory
+# It will also print out what line the error was thrown at and other error info
 
-# You can throw an exception using `die`:
-open 'foo' or die 'Error!'; #=> Error!
+# You can throw your own exceptions using `die`:
+die 'Error!'; #=> Error!
 # Or more explicitly:
 die X::AdHoc.new(payload => 'Error!');
 
+# In Perl 6, `orelse` is similar to the `or` operator, except it only matches
+# `undef` instead of anything evaluating as false.
+# When using `orelse` it will disarm the exception and alias $_ to that failure
+# This will avoid it being automatically handled and printing lots of scary
+# error messages to the screen.
+# We can use the exception method on $_ to access the exception
+open 'foo' orelse say "Something happened {.exception}";
+# This also works, and will print out the object returned even if it's
+# not a failure:
```

Comment:
I think this could be more clear if I described Nil Mu and Failure are all 'undefined type' objects?